src/grelu/io/motifs.py

```diff
@@ -0,0 +1,158 @@
+                name, width, i = None, None, 0
+
+    # Check how many motifs were found
+    print(f"Read {len(motifs)} motifs from file.")
+    if names is not None:
+        missing = set(names).difference(motifs.keys())
+        if len(missing) > 0:
+            print(f"{len(missing)} motifs were not found in the file: {missing}")
+
+    return motifs
+
+
+def read_modisco_report(
+    h5_file: str,
+    group: Optional[str] = None,
+    names: Optional[List[str]] = None,
+    trim_threshold: float = 0.3,
+) -> Dict[str, np.ndarray]:
+    """
+    Reads motifs discovered by TF-MoDISco
+
+    Args:
+        h5_file: Path to an h5 file containing modisco output
+        group: One of "pos" for positive motifs, "neg" for negative motifs or None for all motifs.
+        names: A list containing names of motifs to read. Overrides 'group'.
+        trim_threshold: A threshold value between 0 and 1 used for trimming the PPMs.
+            Each PPM will be trimmed from both ends until the first position for which
+            the probability for any base is greater than or equal to trim_threshold.
+            trim_threshold = 0 will result in no trimming.
+
+    Returns:
+        motifs: A list of motifs as pymemesuite.common.Motif objects
+    """
+    import h5py
+
+    from grelu.interpret.motifs import trim_pwm
+
+    motifs = dict()
+    names_to_read = {
+        "pos": [],
+        "neg": [],
+    }
+
+    # List groups to read
+    if names is not None:
+        for name in names:
+            if name.startswith("pos"):
+                names_to_read["pos"].append(name)
+            elif name.startswith("neg"):
+                names_to_read["neg"].append(name)
+            else:
+                raise ValueError("all names must start with pos or neg")
+
+    else:
+        with h5py.File(h5_file, "r") as f:
+            if group is None:
+                names_to_read["pos"] = list(f["pos_patterns"].keys())
+                names_to_read["neg"] = list(f["neg_patterns"].keys())
+            elif group == "pos":
+                names_to_read["pos"] = list(f["pos_patterns"].keys())
+            elif group == "neg":
+                names_to_read["neg"] = list(f["neg_patterns"].keys())
+            else:
+                raise ValueError("group must be pos, neg or None.")
+
+    # Read CWMs and PPMs from modisco output
+    with h5py.File(h5_file, "r") as f:
+        for group in ["pos", "neg"]:
+            for name in names_to_read[group]:
+
+                ds = f[f"{group}_patterns"][name]
+
+                # Read and normalize PPM
+                ppm = (
+                    ds["sequence"][:] / np.sum(ds["sequence"][:], axis=1, keepdims=True)
+                ).T
+                # Read CWM
+                cwm = ds["contrib_scores"][:].T
+
+                # Trim PPMs based on information content
+                start, end = trim_pwm(
+                    cwm, trim_threshold=trim_threshold, return_indices=True
```

Comment:
I think the trim should be done on the pwm and not cwm? cwm values are not normalized in any way generally

Reply:
I based this on the modiscolite repo where trimming seems to be done on the CWM, e.g.:
https://github.com/jmschrei/tfmodisco-lite/blob/main/modiscolite/report.py#L98
https://github.com/jmschrei/tfmodisco-lite/blob/main/modiscolite/report.py#L236

Is it incorrect?

Reply:
Ok I had a look. Trimming on CWM is fine.

So it looks like within `trim_pwm`, the function recomputes the threshold as 
`trim_thresh = np.max(score) * trim_threshold`
This is a bit unexpected to me, as the threshold should be used as is. Originally, I imagine it's for PWMs where the range is fixed (probs from 0 to 1). 

My suggestion would be to compute exact threshold outside function call

Reply:
Actually, I think the threshold is meant to be calculated on the log-odds matrix, not on the probability matrix. Let me find a reference for that.

Reply:
Here you go: https://bioconductor.org/packages/devel/bioc/vignettes/universalmotif/inst/doc/IntroductionToSequenceMotifs.pdf

Specifically, this section on page 4: 
![image](https://github.com/user-attachments/assets/5d6ab10a-dc1e-4a15-802a-1c95a63c8e4b)

While the context here is motif scanning, not trimming, this is the threshold that we're calculating. It's related to entropy of each position.

Reply:
I see. In either case, the exact threshold calculation should happen outside the `trim_pwm` function. Trim should just take a matrix, some threshold and perform the trimming using that threshold